### PR TITLE
ARROW-11124: [Doc] Update status matrix for Decimal256

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -44,6 +44,8 @@ Data Types
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
 | Decimal128        | ✓     | ✓     |       |            |       |       | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
+| Decimal256        | ✓     | ✓     |       |            |       |       |       |
```
Comment (pitrou) [inline docs/source/status.rst:47]: @quinnj Is Decimal256 supported on Julia?

**Comment by pitrou** [comment]
A JIRA is probably not needed for this.

**Comment by pitrou** [comment]
cc @emkornfield

File: docs/source/status.rst (hunk 1 above)
Comment (quinnj) [reply docs/source/status.rst:47]: Not yet, but I started the work locally, just need to find time to finish it up. It should be pretty easy to at least allow deserializing/serializing, but more work to make the Decimal256 useful for arithmetic, etc.

**Pull request closed without merging**